pkg/container-utils/containerd/containerd.go

```diff
@@ -47,13 +48,26 @@ type ContainerdClient struct {
 	ctx    context.Context
 }
 
-func NewContainerdClient(socketPath string, config *containerutilsTypes.ExtraConfig) (runtimeclient.ContainerRuntimeClient, error) {
+func NewContainerdClient(socketPath string, config *containerutilsTypes.RuntimeConfig) (runtimeclient.ContainerRuntimeClient, error) {
```

Comment:
nit: I think we should avoid passing the whole config because it contains also a socket path, and I would like to avoid creating confusions in the future. I suggest:

```suggestion
func NewContainerdClient(socketPath string, protocol string, config *containerutilsTypes.ExtraConfig) (runtimeclient.ContainerRuntimeClient, error) {
```

Reply:
Updated